Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -453,6 +548,124 @@ private SarifLog RunAnalyzeCommand(string definitionsText, string fileContents)
+            mockFileSystem.Setup(x => x.DirectoryEnumerateFiles(rootDirectory,
+                                                                scanTargetName,
+                                                                It.IsAny<SearchOption>()))
+                                                                    .Returns(new[] { scanTargetPath });
+
+            // Search definitions location and loading
+            mockFileSystem.Setup(x => x.FileExists(searchDefinitionsPath)).Returns(true);
+            mockFileSystem.Setup(x => x.FileReadAllText(It.IsAny<string>()))
+                .Returns<string>((path) =>
+                {
+                    return path == scanTargetPath ?
+                      fileContents :
+                      definitionsText;
+                });
+
+            // Shared strings location and loading
+            mockFileSystem.Setup(x => x.FileReadAllLines(It.IsAny<string>()))
+                .Returns<string>((path) => { return GetSharedStrings(); });
+
+            mockFileSystem.Setup(x => x.FileWriteAllText(It.IsAny<string>(), It.IsAny<string>()))
+                .Callback(new Action<string, string>((path, logText) => { sarifOutput = logText; }));
+
+            mockFileSystem.Setup(x => x.FileInfoLength(SmallTargetName)).Returns(fileContents.Length);
+            mockFileSystem.Setup(x => x.FileExists(@$"c:\Test.UnitTests.Sarif.PatternMatcher.Cli.dll")).Returns(true);
+            mockFileSystem.Setup(x => x.AssemblyLoadFrom(It.IsAny<string>())).Returns(Assembly.LoadFrom(dllLocation));
+
+            Program.FileSystem = mockFileSystem.Object;
+
+            string tempFileName = Path.GetTempFileName();
+            string sarifLogFileName = $"{tempFileName}.sarif";
+            SarifLog sarifLog = null;
+            string levels = "Error;Warning;Note";
+
+            try
+            {
+                string[] args = new[]
+                {
+                    "analyze",
+                    scanTargetPath,
+                    $"-d", searchDefinitionsPath,
+                    $"-o", sarifLogFileName,
+                    "--level", levels,
+                    "--rich-return-code",
+                };
+
+                if (runDynamicValidation)
+                {
+                    args.Append("--dynamic-validation");
+                }
+
+                int result = Program.Main(args);
+                result.Should().Be(0);
+
+                sarifLog = JsonConvert.DeserializeObject<SarifLog>(File.ReadAllText(sarifLogFileName));
+            }
+            finally
+            {
+                if (File.Exists(tempFileName))
+                {
+                    File.Delete(tempFileName);
+                }
+
+                if (File.Exists(sarifLogFileName))
+                {
+                    File.Delete(sarifLogFileName);
+                }
+            }
+
+            return sarifLog;
+        }
+
+        private void ValidateFailureLevelByValidationType(
```

Comment:
>ValidateFailureLevelByValidationType [](http://example.com/codeflow?start=21&length=36)

this isn't a great method name. what is a 'validation type'?

are you really talking about a 'type' here (recall that it's a very overloaded term). you are really talking about a rule implementation detail right?

Also you've duplicated the term 'validate' to refer to your testing but it's also our term for the thing under test. Confusing.

What you're actually testing is the behavior of analysis returning (or not) configured rule failure levels, yes?

Want to try for a better name? :) want me to suggest one?